### PR TITLE
badges as per jds design

### PR DESCRIPTION
**Pull request by ritik07**
Display overview of badges:
Screenshot 2023-01-31 at 11 26 54 AM

1. State of badges as per the JDS and Figma design

#Success
Screenshot 2023-01-31 at 11 18 11 AM

#Error
Screenshot 2023-01-31 at 11 17 43 AM

#Warning
Screenshot 2023-01-31 at 11 18 43 AM

#Default
Screenshot 2023-01-31 at 11 25 32 AM

#Disable
Screenshot 2023-01-31 at 11 26 11 AM

2. If the width is more than 400px then the text in the badge will get truncated 
Screenshot 2023-01-31 at 11 20 03 AM

3. New prop "icon" introduced for the badges which can be used to have an inline icon with text content in a badge
Screenshot 2023-01-31 at 11 21 01 AM

4. Size of bagde as per JDS

#small
Screenshot 2023-01-31 at 11 22 28 AM

#medium
Screenshot 2023-01-31 at 11 22 46 AM

#large
Screenshot 2023-01-31 at 11 23 03 AM

### CODE REVIEW

File: src/base/variable.scss [hunk 1]
```diff
@@ -60,6 +60,25 @@ $Whisper3: #ededed;
 $Gray: #8f8f8f;
 $PinkSwan: #b8b8b8;
 
+////////////////////////////////////JDS THEME COLOR////////////////////////////////////
+/// SUCCESS
+$JDSSuccessLightColor: #E9F7E9;
```
Comment (ShahRohan27) [inline src/base/variable.scss:65]: Please check, colors are already available in existing variables.

Example: #E9F7E9 is $ColorFeedbackSuccess20

File: src/components/Badge/Badge.scss [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
-  border: 0.1rem solid $Zambezi;
+  // border: 0.1rem solid $Zambezi;
   color: $WhiteColor;
   font-size: $BaseFontSize + 2;
   white-space: nowrap;
   font-family: $PrimaryFont;
   line-height: 1.8rem;
-  text-transform: uppercase;
-  max-width: 7.9rem;
+  // text-transform: uppercase;
```
Comment (ShahRohan27) [inline src/components/Badge/Badge.scss:16]: Remove commented code if not required

File: src/components/Badge/Badge.scss [hunk 3]
```diff
@@ -92,3 +96,56 @@
...
+.badge-medium {
+  font-size: $BaseFontSize + 4;
+}
+.badge-large {
+  font-size: $BaseFontSize + 8;
+  padding: 0.4rem 0.8rem;
+  line-height: 2.4rem;
+  letter-spacing: -0.005em;
+}
+.badge-background {
```
Comment (ShahRohan27) [inline src/components/Badge/Badge.scss:113]: prefix classes with n-

File: src/base/variable.scss (hunk 1 above)
Comment (ritik07) [reply src/base/variable.scss:65]: Okay, I'll use an existing variable.

File: src/components/Badge/Badge.scss (hunk 2 above)
Comment (ritik07) [reply src/components/Badge/Badge.scss:16]: okay

File: src/components/Badge/Badge.scss [hunk 4]
```diff
@@ -44,28 +44,32 @@
...
+    // border: 0.1rem solid $Zambezi;
   }
   &.nitrozen-badge-info-fill {
     background: $PrimaryColor;
     border: 0.1rem solid $PrimaryColor;
   }
   &.nitrozen-badge-success-fill {
-    background: $SuccessColor;
-    border: 0.1rem solid $SuccessColor;
+    background: $JDSSuccessLightColor;
```
Comment (Zissan) [inline src/components/Badge/Badge.scss:56]: The same properties are repeated, please consider the use of _mixins_ and _include_

File: src/components/Badge/Badge.scss [hunk 5]
```diff
@@ -92,3 +96,56 @@
...
+  overflow: hidden;
+  position: relative;
+  display: inline-block;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  max-width: 400px;
+}
+
+.badge-small-badge-icon {
+  display: flex;
```
Comment (Zissan) [inline src/components/Badge/Badge.scss:133]: mixins can be used here too.

File: src/components/Badge/Badge.tsx [hunk 6]
```diff
@@ -48,23 +54,44 @@ const Badge = (props: BadgeProps) => {
...
 
   return (
-    <div
-      style={style ?? {}}
-      className={`nitrozen-badge ${classes}`}
-      onClick={onClickHandler}
-      {...restProps}
-    >
-      {labelText}
+    <div className={kind === "service" ? "badge-background" : ""}>
```
Comment (Zissan) [inline src/components/Badge/Badge.tsx:76]: Move this expression as 
`const containerClassname = useMemo(() => (kind === "service" ? "badge-background" : ""), [kind])`

**Review by Zissan — APPROVED** [review]

**Comment by Zissan** [comment]
Please add test cases in a separate PR

**Comment by ShahRohan27** [comment]
Why are we using a hardcoded image as background in service mode for badge. Please check this.
.nitrogen-badge-background {
  background: url("./asset/cover.jpeg");
  background-position: left;
  width: 100%;
  @include nitrozen-vertical-horizontal-center;
  padding: 0.8rem;
}
Please check this.

image


@rit

**Pull request merged**